src/inc/til/at.h

```diff
@@ -34,24 +34,24 @@ namespace til
     // gsl::at will do the check again. As will .at(). And using [] will have a warning in audit.
     // This template is explicitly disabled if T is of type gsl::span, as it would interfere with
     // the overload below.
-    template<class T, std::enable_if_t<!details::is_span<T>::value, int> = 0>
-    constexpr auto at(T& cont, const size_t i) -> decltype(cont[cont.size()])
+    template<typename T, typename I>
+    constexpr auto at(T&& cont, const I i) noexcept -> decltype(auto)

```

Comment:
Alright so `decltype(auto)` here (instead of `auto`) makes sure that this can return an `Elem&` (and will not return an `Elem` instead)?

Reply:
Yep! We could also write `constexpr decltype(auto) ...` but trailing types are technically cooler. 😅